### PR TITLE
Add a CI/PR workflow based on GitHub Actions

### PR DESCRIPTION
**Pull request by dscho**
This takes the incredible work of @sgn and adds a tiny cherry on top.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -255,7 +255,8 @@ jobs:
         exit 1
   linux-musl:
     runs-on: ubuntu-latest
-    container: alpine
+    container:
```
Comment (sgn) [inline .github/workflows/main.yml]: No, in that page:

> When you only specify a container image, you can omit the `image` keyword.
```
jobs:
  my_job:
    container: node:10.16-jessie
```

I'll drop this commit

File: ci/git-problem-matcher.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+    "problemMatcher": [
+        {
+            "owner": "git-test-suite",
+            "pattern": [
+                {
+                    "regexp": "^([^ :]+\\.sh):(\\d+): (error|warning|info):\\s+(.*)$",
+		    "file": 1,
```
Comment (sgn) [inline ci/git-problem-matcher.json:8]: mixed tab and space, I fixed it when cherry-picking.

**Comment by sgn** [comment]
I've
- integrated your change,
- skipped 1 fix-up (container.image)
- made 1 fix-up to a separate change

You can see them here:
https://github.com/git/git/compare/master...sgn:github-action-v4

It's running here: https://github.com/sgn/git/actions/runs/72338937

File: .github/workflows/main.yml (hunk 1 above)
Comment (dscho) [reply .github/workflows/main.yml]: Okay, I used the VS Code extension which apparently has an outdated/incomplete JSON schema.

File: ci/git-problem-matcher.json (hunk 2 above)
Comment (dscho) [reply ci/git-problem-matcher.json:8]: Thank you!

**Comment by sgn** [comment]
I suppose the change in my personal branch is in good-enough state to send to list for review.

If you don't have any other suggestion, and this series haven't been sent by 2020-04-07 01:00 UTC ,
I will send it.

It seems like Junio prefered to have this series sent by you.

**Comment by dscho** [comment]
> I suppose the change in my personal branch is in good-enough state to send to list for review.

Yes, I think so, too!

> If you don't have any other suggestion, and this series haven't been sent by 2020-04-07 01:00 UTC ,
> I will send it.

Would you mind rebasing your branch on top of `v2.26.0`? Then I would be able to use it as-is in this here PR.

> It seems like Junio prefered to have this series sent by you.

In contrast, I find your help invaluable, and by now most of the commits that have me as an author are actually co-authored by you and me. I would be happy if you added a `Co-Authored-by:` footer to those, too, and sent the next iteration.

**Comment by sgn** [comment]
Here is the rebased on top of v2.26.0

But, I don't think you could merge it cleanly to GfW since there'll be conflicts in README.

I added CAB to 1 patch that I actual touch.

**Pull request merged**